app/Http/Controllers/API/v1/TripController.php

```diff
@@ -34,33 +35,49 @@ public function createTrip(Request $request): TripResource {
-                                            'destinationArrivalPlanned' => ['required', 'date'],
-                                        ]);
+        $validated = $request->validate(
+            [
+                'category'                  => ['required', new Enum(HafasTravelType::class)],
+                'lineName'                  => ['required'],
+                'journeyNumber'             => ['nullable', 'numeric', 'min:1'],
+                'operatorId'                => ['nullable', 'numeric', 'exists:hafas_operators,id'],
+                'originId'                  => ['required', 'exists:train_stations,ibnr'],
+                'originDeparturePlanned'    => ['required', 'date'],
+                'destinationId'             => ['required', 'exists:train_stations,ibnr'],
+                'destinationArrivalPlanned' => ['required', 'date'],
+                'stopovers'                 => ['nullable', 'array'],
+                'stopovers.*.stationId'     => ['required', 'exists:train_stations,ibnr'],
+                'stopovers.*.arrival'       => ['required_without:stopovers.*.departure', 'date'],
+                'stopovers.*.departure'     => ['required_without:stopovers.*.arrival,null', 'date'],
+            ]
+        );
 
         DB::beginTransaction();
 
         $creator = new ManualTripCreator();
+        $creator->setCategory(HafasTravelType::from($validated['category']))
+                ->setLine($validated['lineName'], $validated['journeyNumber'])
+                ->setOperator(HafasOperator::find($validated['operatorId']))
+                ->setOrigin(
+                    Station::where('ibnr', $validated['originId'])->firstOrFail(),
+                    Carbon::parse($validated['originDeparturePlanned'])
+                )
+                ->setDestination(
+                    Station::where('ibnr', $validated['destinationId'])->firstOrFail(),
+                    Carbon::parse($validated['destinationArrivalPlanned'])
+                );
 
-        $creator->category                  = HafasTravelType::from($validated['category']);
-        $creator->lineName                  = $validated['lineName'];
-        $creator->journeyNumber             = $validated['journeyNumber'];
-        $creator->operator                  = HafasOperator::find($validated['operatorId']);
-        $creator->origin                    = Station::where('ibnr', $validated['originId'])->firstOrFail();
-        $creator->originDeparturePlanned    = Carbon::parse($validated['originDeparturePlanned']);
-        $creator->destination               = Station::where('ibnr', $validated['destinationId'])->firstOrFail();
-        $creator->destinationArrivalPlanned = Carbon::parse($validated['destinationArrivalPlanned']);
+        if (isset($validated['stopovers'])) {
+            foreach ($validated['stopovers'] as $stopover) {
+                $creator->addStopover(
+                    Station::where('ibnr', $stopover['stationId'])->firstOrFail(),
```

Comment:
(self note) use ID - not IBNR! We are going away from the IBNR as a primary key to support more identifier.

Reply:
I'll change this in another PR, because IBNR is used multiple times in existing code here.